pints/toy/_hes1_michaelis_menten.py

```diff
@@ -50,14 +49,75 @@ class Hes1Model(pints.ForwardModel, ToyModel):
            https://doi.org/10.1038/ncomms1496
     """
     def __init__(self, y0=None, implicit_parameters=None):
-        if y0 is None:
-            self.set_initial_conditions(2)
-        else:
-            self.set_initial_conditions(y0)
         if implicit_parameters is None:
             self.set_implicit_parameters([5., 3., 0.03])
         else:
             self.set_implicit_parameters(implicit_parameters)
+        if y0 is None:
+            self.set_initial_conditions(2)
+        else:
+            self.set_initial_conditions(y0)
+
+    def _dfdp(self, state, time, parameters):
+        """ See :meth:`pints.ToyModel.jacobian()`. """
+        m, p1, p2 = state
+        P0, v, k1, h = parameters
+        p2_over_p0 = p2 / P0
+        p2_over_p0_h = p2_over_p0**h
+        one_plus_p2_expression_sq = (1 + p2_over_p0_h)**2
+        ret = np.empty((self.n_states(), self.n_parameters()))
+        ret[0, 0] = h * p2 * p2_over_p0**(h - 1) / (
+            P0**2 * one_plus_p2_expression_sq)
+        ret[0, 1] = 0
```

Comment:
These =0s aren't strictly necessary, but can keep them in if you think it makes it more readable

Reply:
Oh wait, they are if you use np.empty, but you could use np.zeros instead ?

Reply:
I actually like these as it lays out the matrix which means it's easy to check vs what's expected when reading it.

Reply:
Can we still use np.zeros above, just to avoid any errors if we forget to set one explicitly?

Reply:
Yep, done